extensions/amp-mega-menu/amp-mega-menu.md

```diff
@@ -0,0 +1,228 @@
+
+In the example below, the mega menu contains two menu items with toggleable content, as well as a third item that is an external link.
+
+[example playground="true" imports="amp-mega-menu"]
+```html
+<amp-mega-menu height="30" layout="fixed-height">
+  <nav>
+    <ul>
+      <li>
+        <span role="button">Image</span>
+        <div role="dialog">
+          <amp-img src="{{server_for_email}}/static/inline-examples/images/image1.jpg"
+            width="450"
+            height="300"></amp-img>
+        </div>
+      </li>
+      <li>
+        <span role="button">List</span>
+        <div role="dialog">
+          <ol>
+            <li>item 1</li>
+            <li>item 2</li>
+            <li>item 3</li>
+          </ol>
+        </div>
+      </li>
+      <li>
+        <a href="https://amp.dev/">Link</a>
+      </li>
+    </ul>
+  </nav>
+</amp-mega-menu>
+```
+[/example]
+
+### Dynamic content rendering
+
+The content of a mega menu can be fetched dynamically from a CORS JSON endpoint using [amp-list](https://amp.dev/documentation/components/amp-list/) and rendered via [amp-mustache template](https://amp.dev/documentation/components/amp-mustache).
+
+In the following example, content of the mega menu is rendered dynamically by nesting an `<amp-list>` directly inside the component.
+
+[example playground="true" imports="amp-mega-menu,amp-list" template="amp-mustache"]
+```html
+<amp-mega-menu height="60" layout="fixed-height">
+  <amp-list height="350" layout="fixed-height" src="{{server_for_email}}/static/samples/json/product-single-item.json" single-item>
+    <template type="amp-mustache">
+      <nav>
+        <ul>
+          {{#values}}
+          <li>
+            <h4 role="button">{{name}}</h4>
+            <div role="dialog">
+              <amp-img src="{{server_for_email}}{{img}}" width="320" height="213"></amp-img>
+              <p>Price: $<b>{{price}}</b></p>
+            </div>
+          </li>
+          {{/values}}
+        </ul>
+      </nav>
+    </template>
+  </amp-list>
+</amp-mega-menu>
+```
+[/example]
+
+Here is the JSON file that we used:
+
+```json
+{
+  "items": [{
+    "values": [{
+      "id": 1,
+      "img": "/static/samples/img/product1_640x426.jpg",
+      "name": "Apple",
+      "price": "1.99"
+    }, {
+      "id": 2,
+      "img": "/static/samples/img/product2_640x426.jpg",
+      "name": "Orange",
+      "price": "0.99"
+    }, {
+      "id": 3,
+      "img": "/static/samples/img/product3_640x426.jpg",
+      "name": "Pear",
+      "price": "1.50"
+    }]
+  }]
+}
+```
+
+Note that the `<nav>` element must be parented by either the `<amp-mega-menu>` component itself or a `<template>`.
+
+### Responsive design with `<amp-sidebar>`
+
+For viewports that may be too narrow to display the menu items in one row, we recommend placing the menu inside an [amp-sidebar](https://amp.dev/documentation/components/amp-sidebar/) instead and responsively switch between the two components via media query.
```

Comment:
"placing the menu **inside** amp-sidebar"? This statement is misleading. I think you mean "duplicate" the menu items in an `amp-sidebar`, or literally say "use an `amp-sidebar` for the mobile experience, and responsively switch between the two..."